README.md

```diff
@@ -731,7 +731,15 @@ class NotifyBoardAdmin
 end
 ```
 
-Also, `on_failure` callback will receive `operation_failure:` key in the context containing all the errors.
+Additionally, a callback `call` method can receive the operation result as the second positional argument which is optional. This enables powerful introspection for generic callbacks.
+
+```ruby
+class PublishCommentUpdatedEvent
+  def call(params, operation_result, comment:, **)
```

Comment:
I'm not entirely sure. Maybe it will be better to do `def call(operation_result, params, comment:, **)` ? Overall, operation_result contains both params and context and more data so the signature looks redundant, I just have no clue how it should evolve.

Reply:
Good question for sure! I think I'd prefer to avoid extra duplication here so having just `operation_result` can be fine, on the other hand this is some "hidden" knowledge that we need to pass somehow to others, and then assume that it's like this everywhere

Reply:
>on the other hand this is some "hidden" knowledge that we need to pass somehow to others

Normally I would agree here in a case where this method could possibly expect different objects  which interfaces might be tricky/inconsistent. However, operation_result is always going to be the same, having the same interface. So I think it would be fine to not duplicate rest and pass only the result here @pyromaniac

Reply:
Well, it is gonna break backwards compatibility. We can't just drop params/context. Also, I kinda agree with @ston1x  here that all the pluggable components are having exactly the same signature: `(params, **context)` and I kinda don't want to change this.

On the other hand, prechecks are actually accepting only `(**context)`, no params are passed there, so maybe indeed just `(operation_result)` is a good choice. But now we still need a transition period for this.

I think we can check kwargs presence. If there are any - then `(params, **context)`, if only 1 args and nothing else - `(operation_result)`.

WDYT?

Reply:
Done